### ISSUE TITLE
Can't Tail Large File

### ISSUE THREAD

**Issue by IJustWorkHere** [fluent/fluent-bit#3391]
## Bug Report
When trying to use the tail plugin to process log files greater than or equal to 2gb the `lseek` call fails and the file is removed

**Describe the bug**

Files that are greater than or equal to 2gb in size are removed from the tail plugin

**To Reproduce**
- Example log message if applicable:
```
[2021/04/19 15:47:06] [ info] Configuration:
[2021/04/19 15:47:06] [ info]  flush time     | 5.000000 seconds
[2021/04/19 15:47:06] [ info]  grace          | 5 seconds
[2021/04/19 15:47:06] [ info]  daemon         | 1
[2021/04/19 15:47:06] [ info] ___________
[2021/04/19 15:47:06] [ info]  inputs:
[2021/04/19 15:47:06] [ info]      tail
[2021/04/19 15:47:06] [ info] ___________
[2021/04/19 15:47:06] [ info]  filters:
[2021/04/19 15:47:06] [ info]      modify.0
[2021/04/19 15:47:06] [ info] ___________
[2021/04/19 15:47:06] [ info]  outputs:
[2021/04/19 15:47:06] [ info]      stdout.0
[2021/04/19 15:47:06] [ info] ___________
[2021/04/19 15:47:06] [ info]  collectors:
[2021/04/19 15:47:06] [ info] [engine] started (pid=13976)
[2021/04/19 15:47:06] [debug] [engine] coroutine stack size: 98302 bytes (96.0K)
[2021/04/19 15:47:06] [debug] [storage] [cio stream] new stream registered: tail.0
[2021/04/19 15:47:06] [ info] [storage] version=1.1.1, initializing...
[2021/04/19 15:47:06] [ info] [storage] in-memory
[2021/04/19 15:47:06] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/04/19 15:47:06] [error] [C:\projects\fluent-bit-2e87g\plugins\in_tail\tail_file.c:607 errno=22] Invalid argument
[2021/04/19 15:47:06] [debug] [input:tail:tail.0] inode=3377699720649777 removing file name C:\Users\User\Desktop\service\logs\myservice.1.log
```
- Steps to reproduce the problem:
  * Create a file that is exactly 2gb
```
C:\Users\User\Desktop\service\logs>fsutil file createnew myservice.1.log 2147483648
File C:\Users\User\Desktop\service\logs\myservice.1.log is created

C:\Users\User\Desktop\service\logs>dir
 Volume in drive C has no label.
 Volume Serial Number is xxxx-xxxx

 Directory of C:\Users\User\Desktop\service\logs

04/19/2021  05:31 PM    <DIR>          .
04/19/2021  05:31 PM    <DIR>          ..
04/19/2021  05:31 PM     2,147,483,648 myservice.1.log
               1 File(s)  2,147,483,648 bytes
               2 Dir(s)  773,467,725,824 bytes free
```
  * Run Fluentbit w/ the tail plugin
```
C:\Users\User\Desktop\service\fluentbit\fluentbit-1.7.2>bin\fluent-bit.exe -i tail -p path=C:\Users\User\Desktop\service\logs\matchmaker.1.log -o stdout
Fluent Bit v1.7.2
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/04/19 17:28:59] [ info] [engine] started (pid=7700)
[2021/04/19 17:28:59] [ info] [storage] version=1.1.1, initializing...
[2021/04/19 17:28:59] [ info] [storage] in-memory
[2021/04/19 17:28:59] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/04/19 17:28:59] [error] [C:\projects\fluent-bit-2e87g\plugins\in_tail\tail_file.c:607 errno=22] Invalid argument
```
**Expected behavior**
The service should tail a file e.g.
* Create a file that is one byte less than 2gb
```
C:\Users\User\Desktop\service\logs>fsutil file createnew myservice.1.log 2147483647
File C:\Users\User\Desktop\service\logs\myservice.1.log is created

C:\Users\User\Desktop\service\logs>dir
 Volume in drive C has no label.
 Volume Serial Number is xxxx-xxxx

 Directory of C:\Users\User\Desktop\service\logs

04/19/2021  05:33 PM    <DIR>          .
04/19/2021  05:33 PM    <DIR>          ..
04/19/2021  05:33 PM     2,147,483,647 myservice.1.log
               1 File(s)  2,147,483,647 bytes
               2 Dir(s)  773,466,144,768 bytes free
```
* Start the fluentbit process and send a message
```
C:\Users\User\Desktop\service\fluentbit\fluentbit-1.7.2>bin\fluent-bit.exe -i tail -p path=C:\Users\User\Desktop\service\logs\myservice.1.log -o stdout
Fluent Bit v1.7.2
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/04/19 17:33:13] [ info] [engine] started (pid=6592)
[2021/04/19 17:33:13] [ info] [storage] version=1.1.1, initializing...
[2021/04/19 17:33:13] [ info] [storage] in-memory
[2021/04/19 17:33:13] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/04/19 17:33:13] [ info] [sp] stream processor started
[0] tail.0: [1618878923.470883200, {"log"=>"hello world"}]
```

**Screenshots**
None

**Your Environment**

* Version used:  Fluentbit 1.7.2
* Configuration: `bin\fluent-bit.exe -i tail -p path=C:\Users\User\Desktop\service\logs\myservice.1.log -o stdout`
* Environment name and version (e.g. Kubernetes? What version?): N/A
* Server type and version:
  * Lenovo 30BFS4GK0A
  *  Intel64 Family 6 Model 85 Stepping 4 GenuineIntel ~3696 Mhz
* Operating System and version:
  *  Microsoft Windows 10 Pro for Workstations
  * 10.0.19042 N/A Build 19042
* Filters and plugins:
  * tail/stdout
**Additional context**
We are not able to use fluentbit to tail large files

**Comment by fujimotos** [comment]
@IJustWorkHere FYI, I have just submitted a patch #3590 that fixes this issue.

**Comment by fujimotos** [comment]
Already fixed via #3590
